lib/built_value/built_complex_object.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+import 'package:jsonexample/built_value/built_simple_object.dart';
+
+part 'built_complex_object.g.dart';
+
+abstract class BuiltComplexObject
+    implements Built<BuiltComplexObject, BuiltComplexObjectBuilder> {
+  @nullable
+  String get aString;
+
+  @nullable
+  int get anInt;
+
+  @nullable
+  double get aDouble;
+
+  BuiltSimpleObject get anObject;
+
+  BuiltList<String> get aListOfStrings;
+
+  BuiltList<int> get aListOfInts;
+
+  BuiltList<double> get aListOfDoubles;
+
+  BuiltList<BuiltSimpleObject> get aListOfObjects;
+
+  BuiltComplexObject._();
+
+  static Serializer<BuiltComplexObject> get serializer =>
+      _$builtComplexObjectSerializer;
```

Comment:
I would typically put this at the top of the class, makes it a bit easier to find (and easier to ignore, since you mostly don't care about it when using the class). Up to you, though.

If you do decide to move it, please do so across all files.

Reply:
Done for both simple and complex classes.